src/System.Collections.Immutable/tests/ImmutableListBuilderTest.cs

```diff
@@ -178,6 +178,17 @@ public void Remove()
         }
 
         [Fact]
+        public void RemoveAllBugTest()
+        {
+            var builder = ImmutableList.CreateBuilder<int>();
+            var elemsToRemove = new[]{0, 1, 2, 3, 4, 5}.ToImmutableHashSet();
+            foreach(var elem in new[]{0, 1, 2, 3, 4, 5, 6})
+                builder.Add(elem);
```

Comment:
I just stole the code from the original repro given in the bug.  I can test the AddRange approach, I suspect it is consistent.